kerchunk/hdf.py

```diff
@@ -157,7 +162,17 @@ def translate(self, preserve_linked_dsets=False):
             self.store.flush()
             return self.store
         else:
-            store = _encode_for_JSON(self.store)
+            keys_to_remove = []
+            new_keys = {}
+            for k, v in self.store_dict.items():
+                if isinstance(v, zarr.core.buffer.cpu.Buffer):
+                    key = str.removeprefix(k, "/")
+                    new_keys[key] = v.to_bytes()
+                    keys_to_remove.append(k)
+            for k in keys_to_remove:
+                del self.store_dict[k]
```

Comment:
This is the hacky bit and could use some explanations. Even when requesting "v2", zarr makes Buffer objects, and the keys are also wrong?

Reply:
Yeah so two issues here: 

1. the keys we get from hdf are for example `/depth/.zarray` when then need to be `depth/.zarray`
2. we cant jsonify buffers, which is how the internal `MemoryStore` in v3 stores its data. So we need to convert the buffers to bytes to be serialized

Reply:
OK - would appreciate comments on the code saying this.